api/plugins/guided-tour/admin/src/components/Tour/index.js

```diff
@@ -0,0 +1,74 @@
+import React, { memo, useEffect, useCallback, useReducer, useMemo } from 'react';
+import reducer, { initialState } from './reducer';
+import { useRouteMatch } from 'react-router-dom';
+import { Button } from '@buffetjs/core';
+import ReactTour from 'reactour';
+import { get } from 'lodash';
+
+const Tour = () => {
+  const match = useRouteMatch('/plugins/:pluginId');
+  const pluginId = get(match, ['params', 'pluginId'], 'admin');
+
+  const [{ isOpen, tour, actualPlugin, currentStep, totalLength }, dispatch] = useReducer(
+    reducer,
+    initialState
+  );
+
+  const handleClick = useCallback(() => {
+    dispatch({ type: 'TOGGLE_IS_OPEN', pluginId });
+  }, []);
+
+  const steps = useMemo(() => {
+    return Object.values(tour).reduce((acc, current) => {
+      return [...acc, ...current.steps];
+    }, []);
+  }, [tour]);
+
+  useEffect(() => {
```

Comment:
This effect will be run on every render, can you add the `tour, pluginId, actualPlugin` to the dependency array?

Reply:
Done